src/editor/Model/Menu.re

```diff
@@ -0,0 +1,62 @@
+open Oni_Core;
+open Types;
+open UiMenu;
+
+let create = (~effects: option(Effects.t)=?, ()) => {
+  menu: Closed,
```

Comment:
It's not clear to me what `menu` is for - where does it get set to something besides `Closed`?

Reply:
it was previously called `menuType` it represents the menu's state/type I guess it always has to be something or rather its always theoretically something i.e. `Closed/NoMenu`, `QuickOpen` or `CommandPalette` we change it whenever we open a menu. It tells us its type and we record it in state. The idea being that we can render different UI per menu or use it to specify behaviour per menu type.

It's our only way of knowing what the difference is between the quick open being open or the command palettte